app/models/ability.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    def initialize(user)
+      user ||= User.new # Guest user (not logged in)
+  
+      # Define abilities for different roles
+      if user.admin?
+        can :manage, :all # Admins can manage all resources
+      else
+        can :read, :all
+        can :create, Post
+        can :destroy, Post, author_id: user.id
+        can :destroy, Comment, user_id: user.id
```

Comment:
- You have done well so far, but according to the project requirements, a user should be able to delete a comment if it is theirs or if the user is an admin. You seem not to have implemented this functionality, please fix this 😄